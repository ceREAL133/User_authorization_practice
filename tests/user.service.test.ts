const { createUser, findFewUsers } = require('../src/service/user.service');
import User from '../src/model/user.model';

const userPayload = () => ({
  email: 'jane@example.com',
  name: 'Jane',
  age: 15,
  password: '1CubeIsCool!',
  passwordConfirmation: '1CubeIsCool!',
});

describe('user service', () => {
  describe('create user', () => {
    it('should create user', async () => {
      const oldCreate = User.create;
      User.create = jest.fn(() => Promise.resolve(userPayload())) as any;

      let userInput = userPayload();
      const createdUser = await createUser({
        email: 'jane@example.com',
        name: 'Jane',
        age: 15,
        password: '1CubeIsCool!',
        passwordConfirmation: '1CubeIsCool!',
      });

      expect(createdUser.email).toEqual(userInput.email);
      expect(createdUser.name).toEqual(userInput.name);
      expect(createdUser.age).toEqual(userInput.age);
      expect(createdUser.password).toEqual(userInput.password);
      expect(createdUser.passwordConfirmation).toEqual(
        userInput.passwordConfirmation
      );

      User.create = oldCreate;
    });
  });
});
