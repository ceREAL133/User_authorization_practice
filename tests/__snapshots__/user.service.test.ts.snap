// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user service create user should create user 2 1`] = `
Object {
  "age": 15,
  "email": "jane@example.com",
  "name": "Jane",
  "password": "1CubeIsCool!",
  "passwordConfirmation": "1CubeIsCool!",
}
`;
